### ISSUE TITLE
Marketing Effort for new CE version

### ISSUE THREAD

**Issue by vicos59** [geodesicsolutions-community/geocore-community#64]
This is track what has been done, or needs to be done, to market the new CE version.

Already completed:

1. New Facebook page at https://www.facebook.com/GeoCoreCE
2. https://www.hotscripts.com/listing/geo-classifieds-ads-software/ reported to site owner as obsolete. Submitted new listing for CE.
3. Updates posted to https://indyfor.com/threads/200/ ... which is thread that brought many people together after the company went offline.
4. Direct messaged several people who had posted on the old Geodesic Solutions Facebook page looking for help.

Ongoing tasks:

1. Monitor Google Analytics and Search Console to track visitors and tweak sites as required. There are quite a few showing up in top 10 search results.
2. Continue search for active references to old version/company and post updates.
3. New Logo.

**Comment by WOs8lfTkz3E5g7I4Lm** [comment]
![search results](https://user-images.githubusercontent.com/6634041/153090363-993f2006-a2b4-48be-a63c-a81a4dd67b04.JPG)

GeoCore CE FB page showing at #2 on DuckDuckGo, https://geodesicsolutions.org/ is at #9 
Neither make the top 10 on Google.

**Comment by WOs8lfTkz3E5g7I4Lm** [comment]
@jonyo Did you add the .org site to your Google Search Console yet? If so, would you add me as a user? I received the invite for Analytics.

**Comment by WOs8lfTkz3E5g7I4Lm** [comment]
Attempting to claim ownership of this listing:

https://www.topscriptsdirectory.com/scripts-directory/listing/geocore-for-classified-ads-or-auctions-or-both-professional-classifieds-and-auctions-software

**Comment by jonyo** [comment]
> @jonyo Did you add the .org site to your Google Search Console yet? If so, would you add me as a user? I received the invite for Analytics.

Yes you added on search console, it shows you have full access:
<img width="1398" alt="Screen Shot 2022-02-09 at 07 33 16" src="https://user-images.githubusercontent.com/2767294/153211807-916a783b-2307-47a6-89eb-6c56bc249ea5.png">

**Comment by WOs8lfTkz3E5g7I4Lm** [comment]
@jonyo OK, we're good on the search console. Thanks!

**Comment by WOs8lfTkz3E5g7I4Lm** [comment]
What do you do think about this? Just a quick-and-done graphic in the better-than-nothing dept. Let me know if you want something different.

![GeoCoreCE 1](https://user-images.githubusercontent.com/6634041/153692211-47118af3-a4c0-4e27-8b86-bc02fbcbc217.png)

**Comment by jonyo** [comment]
My first thought is we have to be careful about script fonts like that, it might not be easy to tell what it is for international folks.  I'd be fine with replacing the graphics with just text...  Use Roboto or something like that.  What you think?

Also right now the GeoCore almost disappears because I'm using github with dark theme 😆...

**Comment by WOs8lfTkz3E5g7I4Lm** [comment]
`almost disappears because I'm using github with dark theme`

Yeah, that's pretty much for use on a white background. I'll take another stab it. The idea was to use a serif vs a san serif font so they stood apart from one another.

**Comment by WOs8lfTkz3E5g7I4Lm** [comment]
@jonyo I pulled the website code using Github Desktop and changed resources.php. When I went to push it back up, it told me I had to create a fork due to privs, which I did. Maybe you can check it out to see if I did this right. Just made some minor changes before I do anything extensive.

I used Atom editor. I installed Visual Studio Code, but it bitches about not finding git even though I installed Desktop and CLI.

**Comment by jonyo** [comment]
> `almost disappears because I'm using github with dark theme`
> 
> Yeah, that's pretty much for use on a white background. I'll take another stab it. The idea was to use a serif vs a san serif font so they stood apart from one another.

👍 I liked the alternate color as well, I do something similar on one of my "fun projects".

**Comment by jonyo** [comment]
> @jonyo I pulled the website code using Github Desktop and changed resources.php. When I went to push it back up, it told me I had to create a fork due to privs, which I did. Maybe you can check it out to see if I did this right. Just made some minor changes before I do anything extensive.

For that part, on the wiki page, check out the "how to make a pull request" link.  It sounds like you are almost there now though.

Let me know if you are having trouble with a specific spot.

> 
> I used Atom editor. I installed Visual Studio Code, but it bitches about not finding git even though I installed Desktop and CLI.

That's odd about VSC...  Atom is good to, I know a co-worker used it for a long time.  I used to use Sublime, before that Eclipse (like 10 years ago)...  If you get it set up, it is one of the better ones with xdebug integration.  For the longest time I used Sublime and had VSC installed and would fire it up any time I needed to use xdebug.  But as time went on vsc kept getting better, and at some point I realized it now had all the features I commonly used in my own workflow so I switched over.  I think the majority of engineers that work at my "daytime job" have switched to it as well.  But I digress...

**Comment by WOs8lfTkz3E5g7I4Lm** [comment]
Github Desktop installs git.exe but it is buried way down the tree and they do not add it to the path. Corrected that.

I installed all of the tools and plugins and I am installing the Docker image now. Should be able to figure this out in the next day or two.

I opened a Discussion to post all of this how to stuff on setting up an IDE. Maybe the wiki is better. Not critical, just thinking ahead.

**Comment by jonyo** [comment]
> Github Desktop installs git.exe but it is buried way down the tree and they do not add it to the path. Corrected that.

Ah, my experience is on a mac.  Funny that VSC is made by MS but is complicated to get set up and working on Windows 😆 

> I installed all of the tools and plugins and I am installing the Docker image now. Should be able to figure this out in the next day or two.

Great, keep me posted!

> I opened a Discussion to post all of this how to stuff on setting up an IDE. Maybe the wiki is better. Not critical, just thinking ahead.

Sounds good, both places use the same formatting so it would be easy to copy it into the wiki later if we want.

**Comment by WOs8lfTkz3E5g7I4Lm** [comment]
Now I am stuck. Where you say:

```
Run composer install the first time inside the container:

docker-compose exec --user www-data web composer install

```
I don't understand which CLI I am to use. If I just go to local directory where Github stored the source files I pulled and from a windows command shell execute the docker-compose command, this is what happens:

![1](https://user-images.githubusercontent.com/6634041/153734911-8f799dd1-3cd1-4f8a-b87f-ddde178af8b1.JPG)

These are the Docker Containers I have:

![2](https://user-images.githubusercontent.com/6634041/153734961-f312945c-c4ab-4b57-bdde-af606132a5ae.JPG)

I have no idea what the 2 with the funky names are for. 

If I use the 'CLI' option in Docker next to 'geocore-community-web-1' and run the docker-compose command, it tells me the command is not found. The GeoCore files are there:

![3](https://user-images.githubusercontent.com/6634041/153735016-485d1115-68f6-4aac-8fde-fca2669eaa3b.JPG)

BTW, Are the src files in the Docker image the same as the files that GitHub Desktop pulled to my local PC?  I assume the way this is supposed to work is that I use git to pull the files, make the edits in VSC, then run the changes in the Docker image for testing/debug? I just didn't see the part where the files git pulled are associated with the image. Where is the image I just installed getting the src files from?

**Comment by jonyo** [comment]
> Now I am stuck. Where you say:
> 
> ```
> Run composer install the first time inside the container:
> 
> docker-compose exec --user www-data web composer install
> ```
> 
> I don't understand which CLI I am to use. If I just go to local directory where Github stored the source files I pulled and from a windows command shell execute the docker-compose command, this is what happens:

Hmm, it seems like, looking at all the screenshots, something went sideways when setting it up.

The one showing docker desktop was helpful: the 2 other things running appear to be copies of the geocore-ce (notice the name `geocore-comm...` after each one).

This is what it looks like for me for comparison:

<img width="1270" alt="Screen Shot 2022-02-12 at 21 55 41" src="https://user-images.githubusercontent.com/2767294/153737962-eaf94e49-0bd2-4008-ae93-374729f07a8e.png">

The other "top level" ones you see there are other projects - but notice they all say they are not running.

So on yours:

> ![2](https://user-images.githubusercontent.com/6634041/153734961-f312945c-c4ab-4b57-bdde-af606132a5ae.JPG)

See how the top level one only has 1 thing under it?  There should be 3 - one for the web (this is what makes localhost:8080 show geocore), one for the database, and one for a database admin tool if you need it called adminer (it's kinda like phpmyadmin).

But you only have 1 - the web one.  It won't work since it has no database set up to connect to.

> I have no idea what the 2 with the funky names are for.

I think they were created somehow by maybe running a command more than once and it failed or something?  Not really sure, but maybe try deleting them if it will let you.

> If I use the 'CLI' option in Docker next to 'geocore-community-web-1' and run the docker-compose command, it tells me the command is not found. The GeoCore files are there:
> 
> ![3](https://user-images.githubusercontent.com/6634041/153735016-485d1115-68f6-4aac-8fde-fca2669eaa3b.JPG)

Yeah you run the docker-compose commands from your windows power shell or github cli ETC. like you tried the first time earlier up.

So here's what to do... First, try just deleting all the containers.

Now, you need a command line that is running on your local computer, not in a container, but has all the normal "linux like" commands.  In other words, if you use `ls` it should work and show you the files in your folder.

Actually, maybe you can run it from the command line that is part of VSC...  That is what I do on mac, I do not know if windows version of VSC has it but if it does, use that.  And if you do, then it starts you out in the folder you need to be already!  :)

This is what it looks like for me:

<img width="1309" alt="Screen Shot 2022-02-12 at 22 26 50" src="https://user-images.githubusercontent.com/2767294/153738599-12d045d7-e8c0-4d8a-8dd7-5cbd4e960b73.png">

See at the bottom, look for where it says `terminal` and click that.  Then that is where to type the CLI stuff in (maybe if it works similar in windows).

I haven't set up windows in a while, but if you continue to run into issues I can certainly pull it up on my gaming rig running windows and walk through it so I can give you more tailored instructions.  I don't know when I'll have time but maybe during a lunch break or something coming up next week.

So anyways, if you get a command line or terminal in VSC...

Now run `docker-compose build` - and watch for any errors.  While it is going don't touch docker desktop.

If you do see any errors, post them here and pause.  Errors at this step will mean the rest won't work.

If it completes, then run `docker-compose up` in the same place.  Note the `-` this time!  That is different from the readme instructions, because I think maybe it could be the reason for your problems.  So run that.

I put the other version of the command because a while back I started getting a notice saying that `docker-compose up` is replaced by the other (without a dash).  Maybe they decided it was too confusing or something so went back to the dash version...  Or maybe I got it wrong 😆 

After the first time `docker-compose up` is run, after that it should be able to be controlled in docker desktop.  But you have to run the `build` and the `up` commands the first time to have it all set up correctly.

If it helps, here is what it looks like when I run `docker-compose up` so you know if it looks way different, something is wrong:


https://user-images.githubusercontent.com/2767294/153738836-d058584a-c5b8-437c-8795-a4f732add01e.mov



> 
> BTW, Are the src files in the Docker image the same as the files that GitHub Desktop pulled to my local PC?
>  I assume the way this is supposed to work is that I use git to pull the files, make the edits in VSC, then run the changes in the Docker image for testing/debug? I just didn't see the part where the files git pulled are associated with the image. Where is the image I just installed getting the src files from?

Yes, it kind of sets up some magic link thing.  If you change a file it is almost instantaneously going to be reflected in the container.  I have not looked too close at how it is set up but I believe it actually is the same files, like it is a mapped network drive inside the container or something like that.

**Comment by jonyo** [comment]
By the way, once you have it working...  You can do the same thing for the `website` project.  BUT before you do, be sure to stop the geocore-ce project.  Just stop it, don't delete it.  Because they use the same ports, it will complain if you try to run them at the same time.

Also for the website you don't need the last step because it does not currently use composer.  If you do it right, you can see the geodesicsolutions.org on localhost:8080 with any of your customizations in place, that way you can double check your changes before they "go live".  Just be aware, it will not have the wiki so if you click on any of the wiki links they will be 404's.  (The wiki is just not part of the main "website" project)

**Comment by jonyo** [comment]
@WOs8lfTkz3E5g7I4Lm I looked closer at the screenshot you gave, this one:

> ![1](https://user-images.githubusercontent.com/6634041/153734911-8f799dd1-3cd1-4f8a-b87f-ddde178af8b1.JPG)

That error means somehow you got windows line endings inserted into the `contrib/setup` file.  Did you open files with any text editors other than VSC?  I don't know how that would have happened otherwise, unless somehow git was set up weird, I kind of remember options to convert line endings, but it should not use it by default.

Does git desktop show that file has changes?  If so, undo the changes using git desktop.  Right click on the file from within github desktop, select option to discard changes.

**Comment by jonyo** [comment]
@WOs8lfTkz3E5g7I4Lm I realized I slipped into old habits here of including all the extra info!  I think in the open source community it may be a good idea, so others can dive deeper into why such and such was the answer.

But to give you the **tl;dr:** version of "just the fix":  I think there are 2 things going on.

1. The list of containers in your screenshot is all wrong.  Delete the extra containers that have random names.  Then run `docker-compose run` again in the base folder just to be sure, look for errors.  If you see any, stop here and post them.  If no errors, then run `docker-compose up` - this should set it up for the first time.  If that runs successfully, after that in docker desktop it should look more like my list (ignore the other entries, those are for other projects I have installed on my computer):

![1](https://user-images.githubusercontent.com/2767294/153737962-eaf94e49-0bd2-4008-ae93-374729f07a8e.png)

2. Somehow you got "windows line endings" inserted in the `contrib/setup` file, that is causing the errors when you try to run that command (one of the last instructions in the docker instructions).  That could maybe happen depending on git settings, or it could happen if you opened and saved the file using other than an IDE.  So we need to fix that, maybe you already know how to.  If you don't, tell me what git says, whether it shows changes in the file or not.  That will tell us if it is git to blame or something else.

**Comment by vicos59** [comment]
@jonyo Submitted https://github.com/geodesicsolutions-community/website/pull/11 a few days ago. Just a simple one page. I've been using Github for one of my own projects past week, so I think I have a pretty good feel for it.

**Comment by jonyo** [comment]
> @jonyo Submitted https://github.com/geodesicsolutions-community/website/pull/11 a few days ago.

Thanks for the heads up, apparently I need to adjust my notification settings, currently I seem to only be getting notices on replies to things I have interacted with.

By the way did you ever get docker up and running?

**Comment by vicos59** [comment]
I had to put Docker on the backburner because I need to update/move these last 2 domains off my old server by the 27th. I am hopefully wrapping up the 1st one today, which means I move on the classifieds tomorrow. Once that is done and the old machine is unplugged, I am unencumbered.

**Comment by vicos59** [comment]
![Capture](https://user-images.githubusercontent.com/6634041/155886545-821fcb3e-b8eb-4776-a948-da4d8a3b092a.JPG)

**Comment by vicos59** [comment]
Things are picking up on Facebook.

![fb follows](https://user-images.githubusercontent.com/6634041/156930337-a3793b81-a03f-4650-afc0-ff78977e67fd.JPG)

**Comment by vicos59** [comment]
Things picking up even more:

![search impact](https://user-images.githubusercontent.com/6634041/157459028-aaa494e2-69e2-4ebb-bc0b-579d379e6171.JPG)

BTW, I never heard back from any of the Hotscripts like sites. I imagine they have been abandoned and are on autopilot. Now they just sit there and pollute the search engine results.

**Comment by jonyo** [comment]
> Things picking up even more (up 50% from previous month):

If you compare to this time last year, things are up 1 billion percent!  Or to get more technical, they are up `NaN`% or maybe `Divide by 0 error` percent!  😂 

But seriously, that's great!  It's heading in the right direction thanks to your help.  👍

**Comment by vicos59** [comment]
We're now the proud parents of:

https://geocore.app/

**Comment by jonyo** [comment]
> We're now the proud parents of:
> 
> https://geocore.app/

Nice!  

So umm, why are we the proud parents?  😂  I mean what was the idea behind getting it?  Just to have a shorter domain or so that geocore is in the name or what?

**Comment by vicos59** [comment]
I was just tending to my other domains and the idea came to me. No biggie if you don't ever want to use it in the future. My thinking is that eventually you want to move away from 'Geodesic Solutions' since that is the old company and has nothing to do with the product.

Plus .app is the new thing and geocore.app is short and sweet. For now, it just points to the .org site.

**Comment by vicos59** [comment]
I don't see any need for marketing above and beyond what has been done to date. As far as all of those old Script Index site, like hotscripts, listing the old software, I think we're just SoL. They appear to be abandoned or in archive mode and no one responded to my inquiries to update the listing.

**Comment by vicos59** [comment]
Actually, after looking at the Google results for "geocore classifieds" I see that we need to change the page title for https://geodesicsolutions.org to get better results. Perhaps, "GeoCore Classifieds & Auction Software - Community Edition" which matches the Facebook page.

I also submitted feedback to Google regarding hotscripts and suggested they remove the listing priority since it is not accurate.

![Capture](https://user-images.githubusercontent.com/6634041/185764552-1ec1eca0-fa46-4251-8553-83ae88ea1dfd.JPG)

**Comment by vicos59** [comment]
Reopen....

**Comment by vicos59** [comment]
@jonyo I sent you a Pull Request for some minor changes to the website to help us get better search ranking. 

Side note: Our Facebook page has disappeared from the first page results. Hmmmm. It was there before.

I reported the old page https://www.facebook.com/GeodesicSolutions/ and indicated that it was permanently closed. We'll see if FB actually does anything.

**Comment by jonyo** [comment]
> @jonyo I sent you a Pull Request for some minor changes to the website to help us get better search ranking. 
> 
> Side note: Our Facebook page has disappeared from the first page results. Hmmmm. It was there before.
> 
> I reported the old page https://www.facebook.com/GeodesicSolutions/ and indicated that it was permanently closed. We'll see if FB actually does anything.

Yeah I saw, thanks!

I don't currently have a way to push changes to the site, my old laptop batteries went out and I'm currently waiting on new computer.  With chip shortage the wait is supposed to be 2 to 3 weeks.

Once it gets here I'll be able to check the changes and update the site.  I'm replying on mobile right now 😂

**Comment by vicos59** [comment]
I think it is safe to close this now. Anyone who still uses GeoCore must know about the project by now.
